README.md

```diff
@@ -3,4 +3,356 @@
+is setup correctly first.
+
+Once your Org is configured you can login and set it as the default Dev Hub with the following command:
+
+```sh
+sf login org --alias PythonOrg --set-default-dev-hub --set-default
+```
+
+This will make **PythonOrg** the default Dev Hub for subsequent commands.
+
+Then create a scratch org:
+
+```sh
+sfdx force:org:create \
+  --definitionfile config/project-scratch-def.json \
+  --setalias PythonScratch \
+  --setdefaultusername
+```
+
+This will make it so that when you run a Function, it will connect to and use the **PythonScratch** org.
+
+### Connect Your Compute Environment
+
+Login to Salesforce Functions with the same credentials you used to connect your Dev Hub org.
+
+```sh
+sf login functions
+```
+
+Then you will be able to create the compute environment and associate it with the **PythonScratch** org we created while
+connecting your org.
+
+```sh
+sf env create compute \
+  --connected-org PythonScratch \
+  --alias PythonCompute
+```
+
+This will make it so that when you deploy a Function, it will be deployed to the **PythonCompute** environment linked to 
+your scratch org.
+
+### Assign Permissions
+
+The default Python project you'll generate requires `read` access to the `Account` object in your scratch org.  Create 
+a file named `force-app/main/default/permissionsets/Functions.permissionset-meta.xml` in your SFDX project and add the following content:
+
+```xml
+<?xml version="1.0" encoding="UTF-8"?>
+<PermissionSet xmlns="http://soap.sforce.com/2006/04/metadata">
+    <description>Permissions for Salesforce Functions to access Salesforce org data</description>
+    <hasActivationRequired>true</hasActivationRequired>
+    <label>Functions</label>
+    <objectPermissions>
+        <allowCreate>false</allowCreate>
+        <allowDelete>false</allowDelete>
+        <allowEdit>false</allowEdit>
+        <allowRead>true</allowRead>
+        <modifyAllRecords>false</modifyAllRecords>
+        <object>Account</object>
+        <viewAllRecords>false</viewAllRecords>
+    </objectPermissions>
+</PermissionSet>
+```
+
+Upload this permission set to your org:
+
+```sh
+sf deploy metadata --ignore-conflicts
+```
+
+Then assign the permissions to the `Functions` profile with
+
+```sh
+sfdx force:user:permset:assign -n Functions
+```
+
+## Create and Run a Python Function Locally
+
+### Generate the Python Function
+
+From the SFDX project root, run:
+
+```sh
+sf generate function \
+  --language python \
+  --function-name hellopython
+```
+
+The remaining commands will be executed within the newly generated project folder. Change to that folder by running:
+
+```sh
+cd functions/hellopython
+```
+
+You should also create a `.gitignore` file in the function directory with the following contents:
+```gitignore
+venv
+__pycache__
+```
+
+The following command will do this for you:
+```sh
+echo "venv\n__pycache__" > .gitignore
```

Comment:
This needs to be `echo -e` otherwise the `\n` is interpreted as a literal, not the newline character, resulting in:

```
$ cat .gitignore 
venv\n__pycache__
```

I'll open a PR to fix.

Reply:
Addressed in #34.